frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -299,25 +328,49 @@ export const tabsReducer = createReducer<DashboardState>(
     });
 
     builder.addCase(
-      moveDashCardToTab,
-      (state, { payload: { dashCardId, destTabId } }) => {
-        const { dashId } = getPrevDashAndTabs({ state });
-        if (dashId === null) {
-          throw Error(
-            `MOVE_DASHCARD_TO_TAB was dispatched but dashId (${dashId}) is null`,
-          );
-        }
+      _moveDashCardToTab,
+      (state, { payload: { dashCardId, destinationTabId } }) => {
         const dashCard = state.dashcards[dashCardId];
+        const dashboardId = checkNotNull(state.dashboardId);
```

Comment:
For consistency and to avoid repetition how about we use the `getPrevDashAndTabs` function used in other reducer cases in this file?

```typescript
const { dashId, prevDash } = getPrevDashAndTabs({ state })
```

We can also add the `checkNotNull` assertion inside its body to avoid repeating it, since we expect it to be non-null in all cases.

Reply:
Had to revert putting `checkNotNull` in `getPrevDashAndTabs` as it made [this test](https://github.com/metabase/metabase/blob/9020a5cc254adf3828547bd26d0a2d495436b28c/e2e/test/scenarios/dashboard/x-rays.cy.spec.js#L175-L194) fail :(